### PR TITLE
Updating text to be aligned with the latest spec

### PR DESCRIPTION
**Pull request by joaocc**

### CODE REVIEW

File: contributors/design-proposals/storage/mount-options.md [hunk 1]
```diff
@@ -72,7 +72,8 @@ metadata:
 provisioner: kubernetes.io/glusterfs
 parameters:
   type: gp2
-  mountOptions: "auto_mount"
+mountOptions:
+  - auto_mount
```
Comment (nikhita) [inline contributors/design-proposals/storage/mount-options.md]: This makes sense to me but I'll let @gnufied (author of the design proposal) lgtm this.

**Comment by joaocc** [comment]
I signed it

**Pull request merged**